jinahub/indexers/searcher/vector/AnnoySearcher/__init__.py

```diff
@@ -70,18 +69,22 @@ def _load_index(self, ids, vecs):
         self._indexer.build(self.num_trees)
 
     @requests(on='/search')
-    def search(self, docs: DocumentArray, **kwargs):
-        if not hasattr(self, 'indexer'):
+    def search(self, docs: DocumentArray, parameters: Dict, **kwargs):
+        if not hasattr(self, '_indexer'):
             self.logger.warning('Querying against an empty index')
             return
 
-        for doc in docs.traverse_flat(self.traverse_path):
+        traversal_paths = parameters.get(
+            'traversal_paths', self.default_traversal_paths
+        )
+
+        for doc in docs.traverse_flat(traversal_paths):
             indices, dists = self._indexer.get_nns_by_vector(
                 doc.embedding, self.top_k, include_distances=True
             )
             for idx, dist in zip(indices, dists):
                 match = Document(id=self._ids[idx], embedding=self._vecs[idx])
-                match.score.value = 1 / (1 + dist)
+                match.scores['distance'] = 1 / (1 + dist)
```

Comment:
just curious why changing this to 'distance'

Reply:
this name should be overridable or at least depending on the Annoy metric (can it be euclidean or cosine?

Reply:
'distance' is universal. In this PR I just adapted it to work with the new `.scores` API. I will open a PR to modify all VectorSearchers to  allow configurable distance keys. https://github.com/jina-ai/executor-indexers/issues/26